xcp_d/workflow/anatomical.py

```diff
@@ -573,84 +596,125 @@ def init_anatomical_wf(
+                name="rh_surface_apply_warpfield",
                 mem_gb=mem_gb,
                 n_procs=omp_nthreads,
+            )  # TM
+
+            fs_std_mesh_L = str(
+                get_template(
+                    template="fsaverage",
+                    hemi="L",
+                    space=None,
+                    density="164k",
+                    desc=None,
+                    suffix="sphere",
+                )
             )
 
-            # collect the fsLR mid, pial, wm surfs per hemi and per res (native or 32k)
-            join_lh_native_warped_surfs = pe.JoinNode(
-                niu.Merge(1),
-                name="join_lh_native_warped_surfs",
-                joinsource="lh_surface_apply_warpfield",
-                joinfield="in1",
-                mem_gb=mem_gb,
-                n_procs=omp_nthreads,
+            fs_std_mesh_R = str(
+                get_template(
+                    template="fsaverage",
+                    hemi="R",
+                    space=None,
+                    density="164k",
+                    desc=None,
+                    suffix="sphere",
+                )
             )
 
-            join_rh_native_warped_surfs = pe.JoinNode(
-                niu.Merge(1),
-                name="join_rh_native_warped_surfs",
-                joinsource="rh_surface_apply_warpfield",
-                joinfield="in1",
-                mem_gb=mem_gb,
-                n_procs=omp_nthreads,
+            fs_L2fsLR = str(
+                get_template(
+                    template="fsaverage",
+                    hemi="L",
+                    density="164k",
+                    desc="std",
+                    suffix="sphere",
+                )
+            )
+
+            fs_R2fsLR = str(
+                get_template(
+                    template="fsaverage",
+                    hemi="R",
+                    density="164k",
+                    desc="std",
+                    suffix="sphere",
+                )
+            )
+
+            surface_sphere_project_unproject_lh = pe.Node(
+                SurfaceSphereProjectUnproject(
+                    sphere_project_to=fs_std_mesh_L,
+                    sphere_unproject_from=fs_L2fsLR,
+                    ),
+                name='surface_sphere_project_unproject_lh'
+            )
+            surface_sphere_project_unproject_rh = pe.Node(
+                SurfaceSphereProjectUnproject(
+                    sphere_project_to=fs_std_mesh_R,
+                    sphere_unproject_from=fs_R2fsLR,
+                    ),
+                name='surface_sphere_project_unproject_rh'
             )
 
-            join_lh_32k_surfs = pe.JoinNode(
-                niu.Merge(1),
-                name="join_lh_32k_surfs",
-                joinsource="lh_surface_apply_warpfield",
-                joinfield="in1",
+            # collect the fsLR32k mid, pial, wm surfs per hemi
+
+            lh_32k_surf_wf = pe.MapNode(

```

Comment:
In general if there are only a few things to iterate over, it's better to make explicitly-named nodes instead of using a mapnode. It's more verbose but easier to track down errors and easier to connect their outputs to datasinks